author_api/views.py

```diff
@@ -73,130 +56,152 @@ def get(self, request, *args, **kwargs):
         except:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
-# GET /author/friends/:id
-class GetAuthorFriends(ListAPIView):
-    """
-    Returns a JSON object containing a friend and a list of friendors
-
-    Expected Return JSON:
-        {
-            friend:"username"
-            friendors: [
-                "friendor_username",
-                ...
-            ]
-        }
-    """
+class BaseRelationsMixin(object):
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    queryset = Author.objects.all()
+    lookup_url_kwarg = "id"
+
+class BaseRetrievRelationsView(BaseRelationsMixin, generics.RetrieveAPIView):
+    """Allows read only access for followers/friends"""
+
+    def get_queryset(self):
+        return self.queryset.filter(id = self.kwargs.get(self.lookup_url_kwarg))
+
+class BaseCreateRelationsView(BaseRelationsMixin, generics.CreateAPIView):
+    """Extends CreateAPIView to return a serializer instance upon saving"""
+
+    def perform_create(self, serializer):
+        return serializer.save()
+
+class BaseDeleteRelationsView(BaseRelationsMixin, generics.DestroyAPIView):
+    """Delete followers/friends"""
 
-    def list(self, request, *args, **kwargs):
+    def get_cached_author(self, guid):
+        """Returns a CachedAuthor who will be removed from a relation"""
+        try:
+            return CachedAuthor.objects.get(id = guid)
+        except:
+            # Cached author does not exist on this server
+            raise AuthorNotFound
+
+    def remove_follower(self, author, follower):
+        """
+        Remove the follower and the associated friend for the given author
+        Takes:
+            author: Author model
+            follower: CachedAuthor model
+        """
+        # This will remove the friendship. Follower/Friendship are dependents
+        author.remove_follower(follower)
+
+# Can potentially be moved to a viewset and router
+class CRDFollowers(
```

Comment:
I'd rename this to FollowersView. The CRD part should be evident in the mixins and using that in the name distracts from what it is and doesn't follow any convention we use anywhere else.